050-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,744 @@
+>        3. If deletion failed with any other error this is reported via a condition in the `status.conditions` and the finalizer is not removed.
+---
+
+Similarly to how it is possible to "manage" an existing unmanaged topic in Kafka, we will also provide a mechanism for "unmanaging" a managed topic, via a `strimzi.io/managed=false` annotation.
+This will allow a `KafkaTopic` to be deleted **without** the topic in Kafka being deleted.
+This can be useful operationally, for example to change the `metadata.name` of a `KafkaTopic`.
+
+> An alternative to `strimzi.io/managed=false` would be "explicit deletion" via a `spec.delete` flag.
+> This alternative was rejected because it's incompatible with gitops.
+
+---
+
+#### Example 4: `KafkaTopic` deletion WITHOUT propagation to Kafka
+
+> 1. Assume the topic is currently Kube-managed
+> 
+>     ```yaml=
+>     metadata:
+>       generation: 123
+>       name: foo
+>       finalizer: 
+>       - strimzi.io/topic-operator
+>     spec:
+>       # ...
+>     status:
+>       observedGeneration: 123 # == generation => it is Kube-managed
+>     ```
+> 
+> 2. The user changes the `strimzi.io/managed=false` annotation to `false` so that the topic is no longer Kube-managed.
+> 
+>     ```yaml=
+>     metadata:
+>       generation: 124
+>       name: foo
+>       finalizer: 
+>       - strimzi.io/topic-operator
+>       annotations:
+>       - strimzi.io/managed: false # do not synchronize to Kafka
+>     spec:
+>       # ...
+>     status:
+>       observedGeneration: 123 # != generation => it is Kube-managed
+>     ```
+> 
+> 
+> 3. The user can then delete the resource.
+>     Because of the presence of a `metadata.finalizer` the resource is not actually deleted.
+>     Instead Kube adds the `metadata.deletionTimestamp` field.
+> 
+>     ```yaml=
+>     metadata:
+>       generation: 124
+>       name: foo
+>       finalizer: 
+>       - strimzi.io/topic-operator
+>       annotations:
+>       - strimzi.io/managed: false # do not synchronize to Kafka
+>       deletionTimestamp: 20230301T000000.000
+
+>     spec:
+>       # ...
+>     status:
+>       observedGeneration: 124 # == generation => it is not syncrhonized
+>     ```
+> 
+> 4. The operator notices the `metadata.deletionTimestamp` field.
+>     Since the `spec.magnaged` is `False` it does **not** attempt topic deletion.
+>     1. It removes `strimzi.io/topic-operator` from the `metadata.finalizer`.
+>     2. Kube proceeds to remove the resource.
+
+Note that `strimzi.io/managed: false` also means that a `KafkaTopic` can exist without there being a corresponding topic in Kafka.
+
+---
+
+
+### Operational considerations
+
+#### Changing `metadata.name`
+
+This WILL be supported by:
+
+1. unmanaging the topic (`strimzi.io/managed: false`).
+2. deleting the `KafkaTopic`.
+3. recreating the `KafkaTopic` with a new name.
```

Comment:
I guess this assumes there is `.spec.topicName` set. In other words, you can _rename_ this way the KafkaTopic resource, but not the Kafka topic. I think that might be worth clarifying.